build/teamcity/cockroach/nightlies/microbenchmark_weekly.sh

```diff
@@ -61,17 +73,21 @@ for rev in "${revisions[@]}"; do
   sha_arr+=("$sha")
 done
 
+# Check if the baseline cache exists and copy it to the output directory.
+baseline_cache_path="gs://$BENCH_BUCKET/cache/$GCE_MACHINE_TYPE/$SANITIZED_BENCH_PACKAGE/${sha_arr[1]}"
+if gsutil ls "$baseline_cache_path" &>/dev/null; then
+  mkdir -p "$output_dir/baseline"
+  gsutil -mq cp -r "$baseline_cache_path/*" "$output_dir/baseline"
+  echo "Baseline cache found for ${name_arr[1]}. Using it for comparison."
+  unset "sha_arr[-1]"
+fi
+
 # Builds binaries for the given SHAs.
-BAZEL_SUPPORT_EXTRA_DOCKER_ARGS="-e GOOGLE_EPHEMERAL_CREDENTIALS -e BENCH_PACKAGE -e BUILDS_BUCKET" \
+BAZEL_SUPPORT_EXTRA_DOCKER_ARGS="-e GOOGLE_EPHEMERAL_CREDENTIALS -e SANITIZED_BENCH_PACKAGE -e BENCH_PACKAGE -e BENCH_BUCKET" \
   run_bazel build/teamcity/cockroach/nightlies/microbenchmark_build_support.sh "${sha_arr[@]}"
 
-# Set up credentials (needs to be done after the build phase)
-google_credentials="$GOOGLE_EPHEMERAL_CREDENTIALS"
+# Log into gcloud again (previous credentials are no longer available at this point)
```

Comment:
What do you mean by no longer available?

Reply:
This is a silly workaround, `teamcity-support.sh` has a trap that wipes the credentials file on exit, when the build support script above is executed (it also pulls in teamcity-support.sh to login the gcloud in the container, but wipes the outer file).